File: Codility/03. Time Complexity/FrogJmp/Solution.swift
// https://app.codility.com/programmers/lessons/3-time_complexity/frog_jmp/
// Painless

public func solution(_ X : Int, _ Y : Int, _ D : Int) -> Int {
    return (Y - X) % D == 0 ? (Y - X) / D : ((Y - X) / D) + 1
}

File: Codility/03. Time Complexity/FrogJmp/Solution.py
# https://app.codility.com/programmers/lessons/3-time_complexity/frog_jmp/
# Painless

import math


def solution(X: int, Y: int, D: int) -> int:
    """Compute the jump count as the ceiling of the remaining distance divided by the jump length."""
    # Time: O(1)   Space: O(1)
    return math.ceil((Y - X) / D)


def test() -> None:
    assert solution(10, 85, 30) == 3
    assert solution(10, 10, 5) == 0
    assert solution(1, 5, 2) == 2
